pkg/machine/hyperv/machine.go

```diff
@@ -612,8 +613,22 @@ func (m *HyperVMachine) Start(name string, opts machine.StartOptions) error {
 			m.HostUser.Modified = false
 		}
 	}
+
 	// Write the config with updated starting status and hostuser modification
-	return m.writeConfig()
+	if err := m.writeConfig(); err != nil {
+		return err
+	}
+
+	// Check if gvproxy is still running.
+	// Do this *after* we write config, so we have still recorded that the
+	// VM is actually running - to ensure that stopping the machine works as
+	// expected.
+	_, err = psutil.NewProcess(gvproxyPid)
+	if err != nil {
+		return fmt.Errorf("gvproxy appears to have stopped (PID %d): %w", gvproxyPid, err)
+	}
```

Comment:
Does we actually get here if gvproxy exits right away? I would assume that waiting for the ready socket would juts hang forever or error out there.

Reply:
Reading the code more carefully I this check would need to be before `m.startShares()` as this makes use of SSH and expects a working network connection.

Reply:
Ready socket is a vsock, so it's not fully related.

The main thing I'm concerned about is gvproxy starting, but failing to establish an SSH connection and subsequently exiting, which takes around a second after the VM signalling it's up. I wanted to do this at the end of the function because of this (as much time as possible to wait for gvproxy to do its ssh + backoff logic), but you're right that it should be before setting up shares.

Reply:
Moved code around. I left this here but moved the startShares call later, with its own config write. Seemed safer.